Directory.Build.props

```diff
@@ -26,4 +26,7 @@
     <Optimize>true</Optimize>
   </PropertyGroup>
 
+  <PropertyGroup>
+    <DefineConstants Condition="$([MSBuild]::IsTargetFrameworkCompatible('$(TargetFramework)','netcoreapp3.1'))">$(DefineConstants);NETCORE31COMPATIBLE</DefineConstants>
```

Comment:
@JimBobSquarePants what we need here is really to tell if we are building against .NET Core 3.1+
It's not only about particular API features, I don't have confidence if `Gen2GcCallback` would reliably work on other targets (which are also poorly covered by tests), so I prefer to exclude the whole thing unless we on .NET Core.

Reply:
We should have access to `NETCOREAPP3_1_OR_GREATER` now I believe.

Reply:
Depends if we are using [5.0.2xx SDK](https://github.com/dotnet/sdk/pull/14798) DK on CI. Likely yes, but we need to be 100% sure to a void building the wrong thing. I can't decide based on the build logs.

Reply:
Looks like they're all using 5.0.403. You can see what's installed by following each link in the readme.

https://github.com/actions/virtual-environments